xen/common/coverage/llvm.c

```diff
@@ -44,51 +44,80 @@
     ((uint64_t)'f' << 16) | ((uint64_t)'R' << 8)  | ((uint64_t)129)
 #endif
 
-#define LLVM_PROFILE_VERSION    4
+#if __clang_major__ >= 19
+#define LLVM_PROFILE_VERSION    10
+#define LLVM_PROFILE_NUM_KINDS  3
+#elif __clang_major__ == 18
+#define LLVM_PROFILE_VERSION    9
 #define LLVM_PROFILE_NUM_KINDS  2
+#elif __clang_major__ >= 14 && __clang_major__ <= 17
```

Comment:
nit:
1. I think `&& __clang_major__ <= 17` is not needed
2. if the rest of code is using `>= 18` let's maybe be consistent here